Description: Fix crash on reload budget report.
Origin: upstream, https://github.com/Gnucash/gnucash/commit/238c889da3fadfd1c92502a0eaf4144a899e9e36
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=738113
Last-Update: 2014-10-17
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/engine/gnc-budget.c
+++ b/src/engine/gnc-budget.c
@@ -578,7 +578,8 @@ gnc_budget_get_account_period_value(cons
 Timespec
 gnc_budget_get_period_start_date(const GncBudget *budget, guint period_num)
 {
-    Timespec ts;
+    Timespec ts = {0, 0};
+    g_return_val_if_fail (GNC_IS_BUDGET(budget), ts);
     timespecFromTime64(
         &ts, recurrenceGetPeriodTime(&GET_PRIVATE(budget)->recurrence,
                                      period_num, FALSE));
@@ -588,7 +589,8 @@ gnc_budget_get_period_start_date(const G
 Timespec
 gnc_budget_get_period_end_date(const GncBudget *budget, guint period_num)
 {
-    Timespec ts;
+    Timespec ts = {0, 0};
+    g_return_val_if_fail (GNC_IS_BUDGET(budget), ts);
     timespecFromTime64(
         &ts,  recurrenceGetPeriodTime(&GET_PRIVATE(budget)->recurrence, period_num, TRUE));
     return ts;
